relay-server/src/service.rs

```diff
@@ -365,6 +362,55 @@ impl ServiceState {
     }
 }
 
+fn create_redis_pool(
+    (connection, options): (&RedisConnection, RedisConfigOptions),
+) -> Result<RedisPool, RedisError> {
+    match connection {
+        RedisConnection::Cluster(servers) => {
+            RedisPool::cluster(servers.iter().map(|s| s.as_str()), options)
+        }
+        RedisConnection::Single(server) => RedisPool::single(server, options),
+    }
+}
+
+/// Creates Redis pools from the given `configs`.
+///
+/// If `configs` is [`Unified`](RedisPoolConfigs::Unified), one pool is created and then cloned
+/// for each use case, meaning that all use cases really use the same pool. If it is
+/// [`Individual`](RedisPoolConfigs::Individual), an actual separate pool is created for each
+/// use case.
+pub fn create_redis_pools(configs: RedisPoolConfigs) -> Result<RedisPools, RedisError> {
+    match configs {
+        RedisPoolConfigs::Unified(pool) => {
+            let pool = create_redis_pool(pool)?;
+            Ok(RedisPools {
+                project_configs: pool.clone(),
+                cardinality: pool.clone(),
+                quotas: pool.clone(),
+                misc: pool,
```

Comment:
This is the part that's different from the first PR: we only create one pool, then clone it.